presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/LegacyResourceGroupConfigurationManager.java

```diff
@@ -35,16 +36,41 @@
         NONE
     }
 
+    public static final String NAME = "legacy";
+    public static final String HARD_CONCURRENCY_LIMIT = "hard_concurrency_limit";
+    public static final String MAX_QUEUED_QUERIES = "max_queued_queries";
+
     private static final ResourceGroupId GLOBAL = new ResourceGroupId("global");
 
     private final int hardConcurrencyLimit;
     private final int maxQueued;
 
-    @Inject
-    public LegacyResourceGroupConfigurationManager(QueryManagerConfig config)
+    public LegacyResourceGroupConfigurationManager(int hardConcurrencyLimit, int maxQueued)
+    {
+        checkArgument(hardConcurrencyLimit > 0, "hardConcurrencyLimit must be greater than 0");
+        checkArgument(maxQueued > 0, "maxQueued must be greater than 0");
+        this.hardConcurrencyLimit = hardConcurrencyLimit;
+        this.maxQueued = maxQueued;
+    }
+
+    public static class Factory
+            implements ResourceGroupConfigurationManagerFactory
     {
-        hardConcurrencyLimit = config.getMaxConcurrentQueries();
-        maxQueued = config.getMaxQueuedQueries();
+        @Override
+        public String getName()
+        {
+            return NAME;
+        }
+
+        @Override
+        public ResourceGroupConfigurationManager<?> create(Map<String, String> config, ResourceGroupConfigurationManagerContext context)
+        {
+            String maxQueuedString = requireNonNull(config.get(MAX_QUEUED_QUERIES), "LegacyResourceGroupConfigurationManager must have config max_queued_queries");
+            String hardConcurrencyLimitString = requireNonNull(config.get(HARD_CONCURRENCY_LIMIT), "LegacyResourceGroupConfigurationManager must have config hard_concurrency_liimt");
+            int maxQueued = Integer.parseInt(maxQueuedString);
+            int hardConcurrencyLimit = Integer.parseInt(hardConcurrencyLimitString);
```

Comment:
nit: reorder to keep hardConcurrencyLimit first and then maxQueued to match constructor order